packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -770,6 +799,23 @@ class ScaffoldState extends State<Scaffold> {
           child: config.drawer
         )
       ));
+    } else if (_shouldHandleBackGesture()) {
+      // Add a gesture for navigating back.
+      children.add(new LayoutId(
+        id: _ScaffoldSlot.drawer,
+        child: new Align(
+          alignment: FractionalOffset.centerLeft,
+          child: new GestureDetector(
+            key: _backGestureKey,
+            onHorizontalDragStart: _handleDragStart,
+            onHorizontalDragUpdate: _handleDragUpdate,
+            onHorizontalDragEnd: _handleDragEnd,
```

Comment:
also handle onHorizontalDragCancel, otherwise if the drag is canceled the navigator ends up in a zombie state